src/core/stdcpp/array.d

```diff
@@ -0,0 +1,91 @@
+﻿/**
+ * D header file for interaction with C++ std::array.
+ *
+ * Copyright: Manu Evans 2014 - 2018.
+ * License: Distributed under the
+ *      $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost Software License 1.0).
+ *    (See accompanying file LICENSE)
+ * Authors:   Manu Evans
+ * Source:    $(DRUNTIMESRC core/stdcpp/array.d)
+ */
+
+module core.stdcpp.array;
+
+alias array = std.array;
+
+extern(C++, std):
+
+extern(C++, class) struct array(T, size_t N)
+{
+    alias size_type = size_t;
+    alias difference_type = ptrdiff_t;
+    alias value_type = T;
+    alias reference = ref T;
+    alias const_reference = ref const(T);
+    alias pointer = T*;
+    alias const_pointer = const(T)*;
+
+    alias as_array this;
+
+    extern(D) size_type size() const nothrow @safe @nogc                    { return N; }
```

Comment:
You can just stick an `extern(D):` at the top. Also I would explain in the doc why those are `extern(D)`, as it might be confusing to users/reviewers.

Reply:
Yeah, I thought about that... but then I thought it might be too easy to accidentally mis-attribute any proper extern's.

Reply:
Any proper `extern(C++)` will have it in front of the prototype, won't it ?
I remember doing this for a project of my own and not having to define any `extern(C++)` anyway because everything gets inlined.

Reply:
Yes, `extern(D)` is confusing to me.

Reply:
If these functions are extern(C++), they may conflict with actual C++ implementations of these functions. Locally defined functions should be D-mangled.

But what does it matter, you guys are rejecting it all anyway :/